daemon/cluster/cluster.go

```diff
@@ -239,8 +293,16 @@ func (c *Cluster) startNewNode(forceNewCluster bool, listenAddr, joinAddr, secre
 		reconnectDelay: initialReconnectDelay,
 	}
 	c.node = node
+	c.localAddr = localAddr
+	c.remoteAddr = remoteAddr
 	c.listenAddr = listenAddr
+	c.advertiseAddr = advertiseAddr
 	c.saveState()
+
+	// Do this assignment after saveState so that the result of
+	// the resolution won't be saved persistently.
+	c.localAddr = actualLocalAddr
```

Comment:
On reconnection it seems to be reusing `c.localAddr` and still saving it. I'm not actually sure about saving the `remoteAddr` on disk and reusing it. It is likely only valid on the time of the request. If I take down that node that was used as a remote for a specific node I don't want it to affect that node. If I understand it correct this case may even mean that my node doesn't start anymore.